### PR TITLE
feat: add autoware msg bag converter

### PR DESCRIPTION
**Pull request by hayato-m126**
## Description

Added tool to convert bag containing autoware_auto_msg to autoware_msg

## Tests performed

colcon test

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_msg_bag_converter/dependency.repos [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  # core
+  core/autoware_msgs:
+    type: git
+    url: https://github.com/autowarefoundation/autoware_msgs.git
+    version: main
+  core/autoware_common:
+    type: git
+    url: https://github.com/autowarefoundation/autoware_common.git
+    version: main
+  core/external/autoware_auto_msgs: # TODO(mfc): Remove when autoware_msgs is merged
```
Comment (SakodaShintaro) [inline autoware_msg_bag_converter/dependency.repos:11]: I think it is okay to delete the comment as it seems unnecessary.

File: autoware_msg_bag_converter/package.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>autoware_msg_bag_converter</name>
+  <version>0.1.0</version>
+  <description>TODO: Package description</description>
```
Comment (SakodaShintaro) [inline autoware_msg_bag_converter/package.xml:6]: You should probably make a change because there is still a TODO left.

**Review by SakodaShintaro — COMMENTED** [review]
Thank you very much! I think it's basically fine.
However, let's wait until get approve from mitsudome-san before merging.

File: autoware_msg_bag_converter/package.xml (hunk 2 above)
Comment (hayato-m126) [reply autoware_msg_bag_converter/package.xml]: I fixed with the following commit hash
95a227027c351b1a8980b6f40471158ddcb5bb40

File: autoware_msg_bag_converter/dependency.repos (hunk 1 above)
Comment (hayato-m126) [reply autoware_msg_bag_converter/dependency.repos]: I fixed with the following commit hash
4617d5e081db6686f516894ff61f41f31088ba1d

**Comment by hayato-m126** [comment]
Since there is a dependency on autoware_auto_msg, cut it out as a separate branch instead of putting it in this repository

**Pull request closed without merging**